src/locator/index.ts

```diff
@@ -360,7 +364,15 @@ export function locate(matrix: BitMatrix): QRLocation {
 
   // Now that we've found the three finder patterns we can determine the blockSize and the size of the QR code.
   // We'll use these to help find the alignment pattern but also later when we do the extraction.
-  const { dimension, moduleSize } = computeDimension(topLeft, topRight, bottomLeft, matrix);
+  let dimension: number;
+  let moduleSize: number;
+  try {
+    const computedDimension = computeDimension(topLeft, topRight, bottomLeft, matrix);
```

Comment:
nit: You could use destructuring here.

`({ dimension, moduleSize } = computeDimension(topLeft, topRight, bottomLeft, matrix));`

Reply:
Ah I was trying to do that but didn't have the syntax right. Added in a8ceef6.